### ISSUE TITLE
acf.createWeapon giving an error when trying to create weapon

### ISSUE THREAD

**Issue by BPRJon** [thegrb93/StarfallEx#1733]
I'm trying to spawn an ACF gun with acf.createWeapon using this code:
`acf.createWeapon(chip():getPos(),Angle(),{
    ClassID    =    "HW",
    Caliber    =    75,
    Destiny    =    "Weapons",
    ID    =    "75mmHW"
    })`

and it is giving me back this error:
`starfall/libs_sh/acffunctions.lua:707: attempt to index field 'props' (a nil value)
stack traceback:
	starfall/libs_sh/acffunctions.lua:707: in function 'createWeapon'
	SF:acf_spawn_test.txt:18: in main chunk
`

**Comment by adamnejm** [comment]
Even though this is caused by an extension, I think it should be dealt with here.
[`propUndo`](https://github.com/thegrb93/StarfallEx/blob/master/lua/starfall/libs_sv/prop.lua#L34) should be stored in the instance instead of an upvalue, so extensions can access the setting.

**Comment by legokidlogan** [comment]
> should be stored in the instance instead of an upvalue, so extensions can access the setting.

Absolutely, I'd love to have access to it for [Quad Props](https://github.com/legokidlogan/quad_props) and its sf extension

**Comment by thegrb93** [comment]
There's a lot of other stuff broken in their lib regarding using prop locals. Their lib should probably use its own locals.

**Comment by thegrb93** [comment]
Fixed the instance.data.props.undo being missing. They'll need to update instance.data.props.props though to instance.data.props.propList and use its register function.
